cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -676,6 +677,105 @@ const OPCODE_SIGNATURES: &'static [(
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I16, I8, I8], &[I8], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I16, I16, I16], &[I16], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I16, I32, I32], &[I32], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I16, I64, I64], &[I64], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I16, I128, I128], &[I128], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I32, I8, I8], &[I8], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I32, I16, I16], &[I16], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I32, I32, I32], &[I32], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I32, I64, I64], &[I64], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I32, I128, I128], &[I128], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I64, I8, I8], &[I8], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I64, I16, I16], &[I16], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I64, I32, I32], &[I32], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I64, I64, I64], &[I64], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I64, I128, I128], &[I128], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I128, I8, I8], &[I8], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I128, I16, I16], &[I16], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I128, I32, I32], &[I32], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I128, I64, I64], &[I64], insert_opcode),
+    #[cfg(not(any(target_arch = "x86_64", target_arch = "aarch64")))]
+    (Opcode::SelectSpectreGuard, &[I128, I128, I128], &[I128], insert_opcode),
```

Comment:
I'm a little hesitant to add opcodes to this list where we have to disable all versions of the opcode on all of the architectures that we're actually fuzzing on. I suppose we could add an `insert_select_spectre_guard` function that generates an `icmp` first. It's not great in the long run, since I like your idea of eventually generating this table from the meta instruction definitions. But it would get us some more fuzz test coverage quickly, at least.

Reply:
Yeah, I don't mind adding that right now.

I didn't open issues for `select_spectre_guard` since I wasn't sure, but are we planning on implementing that? Or does that opcode only make sense when paired with something else?

Reply:
👍 Thanks!

I've added the special `insert_select_spectre_guard`, and surprisingly this works for all variations of `select_spectre_guard`.

I'll wait for a decision to be reached on #5206 before opening the "missing lowerings" issues.